web/src/main/java/org/cbioportal/web/util/StudyViewFilterApplier.java

```diff
@@ -125,20 +125,50 @@ public SampleIdentifier apply(Sample sample) {
 
         if (!CollectionUtils.isEmpty(studyViewFilter.getGeneFilters())) {
 
-            Set<String> molecularProfileIds = studyViewFilter.getGeneFilters().stream()
-                    .flatMap(geneFilter -> geneFilter.getMolecularProfileIds().stream()).collect(Collectors.toSet());
+            Set<String> molecularProfileIds = studyViewFilter.getGeneFilters()
+                    .stream()
+                    .flatMap(geneFilter -> {
+                        // TODO: cleanup once fusion/structural data is fixed in database
+                        // until then use mutation profile to fetch fusions
+                        return geneFilter.getMolecularProfileIds().stream().flatMap(molecularProfileId -> {
+                            ArrayList<String> ids = new ArrayList<String>();
+                            ids.add(molecularProfileId);
+                            if (molecularProfileId.endsWith("_fusion")) {
```

Comment:
This is a temporary fix. the _fusion profile is create in the database for studies that still have fusions in the mutation table.